pkg/networkpath/traceroute/common/common_windows.go

```diff
@@ -0,0 +1,184 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package common contains common functionality for both TCP and UDP
+// traceroute implementations
+package common
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"golang.org/x/net/ipv4"
+	"golang.org/x/sys/windows"
+)
+
+var (
+	sendTo   = windows.Sendto
+	recvFrom = windows.Recvfrom
+)
+
+type (
+	// Winrawsocket is a struct that encapsulates a raw socket
+	// on Windows that can be used to listen to traffic on a host
+	// or send raw packets from a host
+	Winrawsocket struct {
+		Socket windows.Handle
+	}
+
+	MatcherFunc func(*ipv4.Header, []byte, net.IP, uint16, net.IP, uint16, uint32) (net.IP, error)
+
+	MismatchError string
+)
+
+// Close closes the raw socket
+func (w *Winrawsocket) Close() {
+	if w.Socket != windows.InvalidHandle {
+		windows.Closesocket(w.Socket) // nolint: errcheck
+	}
+	w.Socket = windows.InvalidHandle
+}
+
+// CreateRawSocket creates a Winrawsocket
+func CreateRawSocket() (*Winrawsocket, error) {
```

Comment:
Unless we have the same function for other platforms, this should be more specific, e.g. `NewWinRawSocket`.